### PR TITLE
ddl: Clean up

### PR DESCRIPTION
**Pull request by zimulala**
remove unnecessary code in table.go
improve the test coverage and clean up tests in ddl_worker_test.go

### CODE REVIEW

File: ddl/table.go [hunk 1]
```diff
@@ -48,13 +48,9 @@ func (d *ddl) onCreateTable(t *meta.Meta, job *model.Job) error {
 	// Check the table.
 	for _, tbl := range tables {
 		if tbl.Name.L == tbInfo.Name.L {
-			if tbl.ID != tbInfo.ID {
```
Comment (hanfei1991) [inline ddl/table.go]: Why remove this ?

File: ddl/ddl_worker_test.go [hunk 2]
```diff
@@ -178,69 +169,70 @@ func (s *testDDLSuite) TestIndexError(c *C) {
...
+			[]*ast.IndexColName{{Column: &ast.ColumnName{Name: model.NewCIStr("c")}, Length: 256}}}, ctx, d)
 	testCreateIndex(c, ctx, d, dbInfo, tblInfo, false, "c1_index", "c1")
-
 	doDDLJobErr(c, dbInfo.ID, tblInfo.ID, model.ActionAddIndex,
-		[]interface{}{false, model.NewCIStr("c1_index"), []*ast.IndexColName{{Column: &ast.ColumnName{Name: model.NewCIStr("c1")}, Length: 256}}}, ctx, d)
-	doDDLJobErr(c, dbInfo.ID, tblInfo.ID, model.ActionDropIndex, []interface{}{1}, ctx, d)
+		[]interface{}{false, model.NewCIStr("c1_index"), 1,
+			[]*ast.IndexColName{{Column: &ast.ColumnName{Name: model.NewCIStr("c1")}, Length: 256}}}, ctx, d)
 
+	// for drop index
```
Comment (hanfei1991) [inline ddl/ddl_worker_test.go]: drop -> dropping

File: ddl/ddl_worker_test.go [hunk 3]
```diff
@@ -147,6 +132,15 @@ func (s *testDDLSuite) TestTableError(c *C) {
 		return nil
 	})
 	c.Assert(err, IsNil)
+
+	// Args is wrong, so creating table is failed.
+	doDDLJobErr(c, 1, 1, model.ActionCreateTable, []interface{}{1}, ctx, d)
+	// Schema ID is wrong, so creating table is failed.
+	doDDLJobErr(c, -1, tblInfo.ID, model.ActionCreateTable, []interface{}{tblInfo}, ctx, d)
+	// Table is existed, so creating table is failed.
```
Comment (hanfei1991) [inline ddl/ddl_worker_test.go]: Table exists

File: ddl/ddl_worker_test.go [hunk 4]
```diff
@@ -147,6 +132,15 @@ func (s *testDDLSuite) TestTableError(c *C) {
 		return nil
 	})
 	c.Assert(err, IsNil)
+
+	// Args is wrong, so creating table is failed.
+	doDDLJobErr(c, 1, 1, model.ActionCreateTable, []interface{}{1}, ctx, d)
+	// Schema ID is wrong, so creating table is failed.
```
Comment (hanfei1991) [inline ddl/ddl_worker_test.go]: fail to create table

File: ddl/ddl_worker_test.go [hunk 5]
```diff
@@ -113,29 +108,19 @@ func (s *testDDLSuite) TestTableError(c *C) {
...
+	// Schema ID is wrong, so dropping table is failed.
+	doDDLJobErr(c, -1, 1, model.ActionDropTable, nil, ctx, d)
+	// Table ID is wrong, so dropping table is failed.
 	dbInfo := testSchemaInfo(c, d, "test")
 	testCreateSchema(c, testNewContext(c, d), d, dbInfo)
+	job := doDDLJobErr(c, dbInfo.ID, -1, model.ActionDropTable, nil, ctx, d)
 
-	job = doDDLJobErr(c, dbInfo.ID, -1, model.ActionDropTable, nil, ctx, d)
-
+	// Table ID or schema ID is wrong, so getting table is failed.
```
Comment (hanfei1991) [inline ddl/ddl_worker_test.go]: fail to get table

File: ddl/ddl_worker_test.go [hunk 6]
```diff
@@ -113,29 +108,19 @@ func (s *testDDLSuite) TestTableError(c *C) {
...
-	job.State = 0
-	tblInfo := testTableInfo(c, d, "t", 3)
-	job.Args = []interface{}{tblInfo}
-	err := d.doDDLJob(ctx, job)
-	c.Assert(err, NotNil)
-	testCheckJobCancelled(c, d, job)
-
-	doDDLJobErr(c, 1, 1, model.ActionDropTable, nil, ctx, d)
-
+	// Schema ID is wrong, so dropping table is failed.
```
Comment (hanfei1991) [inline ddl/ddl_worker_test.go]: fail to drop table

Comment (shenli) [inline ddl/ddl_worker_test.go]: Why remove the model.ActionCreateTable job related code?

File: ddl/table.go (hunk 1 above)
Comment (zimulala) [inline ddl/table.go]: If `tbl.Name.L == tbInfo.Name.L` is true,  the job need to cancel. You can see the comments.

File: ddl/table.go [hunk 7]
```diff
@@ -48,13 +48,9 @@ func (d *ddl) onCreateTable(t *meta.Meta, job *model.Job) error {
...
-				// This table already exists, can't create it, we should cancel this job now.
-				job.State = model.JobCancelled
-				return errors.Trace(infoschema.ErrTableExists)
-			}
-
-			tbInfo = tbl
+			// This table already exists, can't create it, we should cancel this job now.
```
Comment (hanfei1991) [inline ddl/table.go]: This table already exists and can't be created,

**Comment by hanfei1991** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

**Pull request merged**